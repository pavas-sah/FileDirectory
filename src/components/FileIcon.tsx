import React from 'react';

export interface FileIconProps {
  url: string;
  alt?: string;
}

export const FileIcon: React.SFC<FileIconProps> = (props) => {
  return <img src={props.url} alt={props.alt} />;
};
export default FileIcon;
