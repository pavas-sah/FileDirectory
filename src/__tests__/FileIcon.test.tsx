import React from 'react';
import renderer from 'react-test-renderer';
import FileIcon, { FileIconProps } from '../components/FileIcon';

const IconProps: FileIconProps = {
  url: 'some url',
  alt: 'alternate message',
};

describe('File Icon Functional component', () => {
  it('should have props with defined values.', () => {
    const iconComponent = renderer.create(<FileIcon {...IconProps} />);
    expect(iconComponent.toJSON()?.props).toEqual({
      alt: 'alternate message',
      src: 'some url',
    });
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<FileIcon {...IconProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
