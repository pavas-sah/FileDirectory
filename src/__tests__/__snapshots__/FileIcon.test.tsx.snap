// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Icon Functional component matches snapshot 1`] = `
<img
  alt="alternate message"
  src="some url"
/>
`;
